### DISCUSSION TITLE
How to compose a collection from elements defined in other files

### DISCUSSION THREAD

**Discussion by ekille-foreflight** [apple/pkl#838] [Q&A] [answered]
Below `broadStInventory` amends Inventory. In the `cars` Listing, I want to insert 'civic' which is built out in `civic.pkl` (instead of using new like the Rav4)

broadStInventory.pkl
```
amends "Inventory.pkl"
import "civic.pkl"

name = "Broad Street"

cars {
        civic    // is there a way to make this work?
	new {
		name = "Rav 4"
		year = 2007
	}
}
```

Inventory.pkl
```
import "Motorcycle.pkl"
import "Car.pkl"

name: String   // name of inventory
cars: Listing<Car>
motorcycles: Listing<Motorcycle>

class Inventory {
  motorcycles: Listing<Motorcycle>
  cars: Listing<Car>
}
```

civic.pkl
```
import "Car.pkl"

name = "Civic EX"
year = "1997"
```

**Comment by HT154** [comment, marked as answer]
This is very close as-is! The only difference is that civic.pkl must `amend` Car.pkl instead of importing it. Amending a module is conceptually identical to instantiating it via `new`

civic.pkl
```
amends "Car.pkl"

name = "Civic EX"
year = "1997"
```

Separately, the `Inventory` class in your Inventory.pkl module seems odd. The class appears unused and duplicate the module's existing properties. You can likely remove the class entirely. In Pkl, modules _are_ classes.

**Reply by ekille-foreflight** [reply]
That was it! Inventory definitely needed some clean up. Thanks!
